src/async_impl/client.rs

```diff
@@ -565,10 +565,21 @@ impl ClientBuilder {
                         return Err(crate::error::builder("empty supported tls versions"));
                     }
 
+                    #[cfg(feature = "rustls-tls-platform-verifier")]
+                    let verifier = Arc::new(rustls_platform_verifier::Verifier::new());
+                    #[cfg(not(feature = "rustls-tls-platform-verifier"))]
+                    let verifier =
```

Comment:
The `WebPkiServerVerifierBuilder::build()` step yields `Err` if no roots have been passed in, meaning that a whole bunch of tests now panic when testing under `rustls-tls-manual-roots` (without specifying any roots). I've just disabled these tests when testing with this particular feature, which seems like the easiest approach?

Alternatively, we could have slightly more complex setup here which would avoid the error at builder time, but failing earlier actually seems desirable to me.